### PR TITLE
Bug 1826514: Add unit test for snapshot views

### PR DESCRIPTION
**Pull request by a2batic**
Added unit test for:-
1. VolumeSnapshot Page
2. VolumeSnapshot modal
3. Restore Modal

Fixes: https://issues.redhat.com/browse/RHSTOR-946
Signed-off-by: Kanika <kmurarka@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/__mocks__/volume-snapshot-data.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      storageClassName: 'fake-scn',
+    },
+    status: {
+      phase: 'Bound',
+      accessModes: ['ReadWriteOnce'],
+      capacity: {
+        storage: '1Gi',
+      },
+    },
+  } as unknown) as FirehoseResult<K8sResourceCommon | K8sResourceCommon[]>,
```
Comment (vojtechszocs) [inline frontend/packages/ceph-storage-plugin/src/__mocks__/volume-snapshot-data.tsx:87]: I'd suggest to skip typing of `pvcData` variable here and force-cast it to `FirehoseResult< .. >` in the test that imports & uses it.

More generally, typing to `any` or `unknown` and then to a specific type should be avoided.

File: frontend/packages/ceph-storage-plugin/src/__tests__/restore-pvc-modal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  RestorePVCModalProps,
+  RestorePVCModal,
+} from '../components/modals/restore-pvc-modal/restore-pvc-modal';
+import { snapshotData } from '../__mocks__/volume-snapshot-data';
+import { VolumeSnapshotModal } from '../components/modals/volume-snapshot-modal/volume-snapshot-modal';
+
+import Spy = jasmine.Spy;
+
+describe(VolumeSnapshotModal.name, () => {
```
Comment (vojtechszocs) [inline frontend/packages/ceph-storage-plugin/src/__tests__/restore-pvc-modal.spec.tsx:16]: This unit test works with `RestorePVCModal` (?)

```suggestion
describe(RestorePVCModal.name, () => {
```

File: frontend/packages/ceph-storage-plugin/src/__tests__/volume-snapshot.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+} from '../components/volume-snapshot/volume-snapshot';
+
+import { DeleteSnapshot } from '../utils/delete-snapshot-workflow';
+import { RestorePVC } from '../utils/restore-pvc-workflow';
+import { snapshotData, pvcData } from '../__mocks__/volume-snapshot-data';
+
+describe('<VolumeSnapshotList>', () => {
+  let wrapper: ShallowWrapper<TableProps>;
+  let rowArgs: RowFunctionArgs<K8sResourceKind>;
+  beforeEach(() => {
```
Comment (vojtechszocs) [inline frontend/packages/ceph-storage-plugin/src/__tests__/volume-snapshot.spec.tsx:35]: Nit: please put a newline above `beforeEach` call.

File: frontend/packages/ceph-storage-plugin/src/__tests__/volume-snapshot.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    expect(wrapper.find(ListPage).exists()).toBe(true);
+  });
+});
+
+describe('<VolumeSnapshotDetails>', () => {
+  let wrapper: ShallowWrapper<any>;
+  const detailArgs = {
+    obj: snapshotData.items[0],
+  } as any;
+  beforeEach(() => {
```
Comment (vojtechszocs) [inline frontend/packages/ceph-storage-plugin/src/__tests__/volume-snapshot.spec.tsx:117]: Nit: please put a newline above `beforeEach` call.

**Comment by a2batic** [comment]
/retitle Bug 1826514: Add unit test for snapshot views

**Pull request merged**